lib/Network/OAuth2/Server/API.hs

```diff
@@ -423,7 +424,7 @@ serverCreateToken
 serverCreateToken pool user_id userScope reqScope = do
     if compatibleScope reqScope userScope then do
         TokenID t <- liftIO $ storeCreateToken pool user_id reqScope
-        throwError err302{errHeaders = [(hLocation, "/tokens?token_id=" <> T.encodeUtf8 t)]} --Redirect to tokens page
+        throwError err302{errHeaders = [(hLocation, "/tokens?token_id=" <> U.toASCIIBytes t)]} --Redirect to tokens page
```

Comment:
To the end of not accidentally generating broken URIs when t starts being something other than what you expect?